packages/gestalt/src/Checkbox.js

```diff
@@ -207,7 +201,10 @@ const CheckboxWithForwardRef: React$AbstractComponent<Props, HTMLInputElement> =
         </Label>
         {Boolean(image) && <Box paddingX={1}>{image}</Box>}
         {label && (
-          <Box display={labelDisplay === 'hidden' ? 'visuallyHidden' : 'block'}>
+          <Box
+            display={labelDisplay === 'hidden' ? 'visuallyHidden' : 'block'}
+            dangerouslySetInlineStyle={{ __style: { marginTop: '2px' } }}
```

Comment:
why is the 2px needed?

Reply:
@ayeshakmaz 

Without the 2 pixels looks unbalanced. @ponori Thoughts on these 2 pxs?

NO pixels
<img width="391" alt="Screen Shot 2022-09-20 at 1 07 06 AM" src="https://user-images.githubusercontent.com/10593890/191140536-1f0066ff-9f18-4673-9d02-bc2c30d023bc.png">

WITH 2 MARGIN TOP PIXELS
![image](https://user-images.githubusercontent.com/10593890/191140557-1c840a98-6e19-4942-8305-64dd5513b911.png)

Reply:
Yeah, I see what you're saying Alberto... Ayesha, my guess is that this is problematic b/c the spacing doesn't translate to a spacing token. Would be interesting to hear if there was a way to have our cake (proper alignment) and eat it too (utilize tokens).

Reply:
Does this need to be a token? We use little tweaks like this all over the place where mixing elements of different types (icons + text, checkbox + text, etc). I'm personally fine with these nudges as long as we make note of why the additional margin is included in a code comment

Reply:
COMMENT ADDED.
DONE

Reply:
Yeah agreed, don't think it needs to be a token, but definitely needs the comment. Thanks Alberto!